src/basic/terminal-util.c

```diff
@@ -1753,34 +1765,41 @@ int get_default_background_color(double *ret_red, double *ret_green, double *ret
         BackgroundColorContext context = {};
 
         for (;;) {
-                usec_t n = now(CLOCK_MONOTONIC);
+                if (buf_full == 0) {
+                        usec_t n = now(CLOCK_MONOTONIC);
 
-                if (n >= end) {
-                        r = -EOPNOTSUPP;
-                        goto finish;
-                }
+                        if (n >= end) {
+                                r = -EOPNOTSUPP;
+                                goto finish;
+                        }
 
-                r = fd_wait_for_event(STDIN_FILENO, POLLIN, usec_sub_unsigned(end, n));
-                if (r < 0)
-                        goto finish;
-                if (r == 0) {
-                        r = -EOPNOTSUPP;
-                        goto finish;
-                }
+                        r = fd_wait_for_event(STDIN_FILENO, POLLIN, usec_sub_unsigned(end, n));
+                        if (r < 0)
+                                goto finish;
+                        if (r == 0) {
+                                r = -EOPNOTSUPP;
+                                goto finish;
+                        }
 
-                ssize_t l;
-                l = read(STDIN_FILENO, buf, sizeof(buf) - buf_full);
-                if (l < 0) {
-                        r = -errno;
-                        goto finish;
-                }
+                        ssize_t l = read(STDIN_FILENO, buf, sizeof(buf) - buf_full);
```

Comment:
Hmm, this should read into `buf + buf_full`?

Reply:
it indeed should